### PR TITLE
nix: add conditionals for non-flake builds and support channels

### PR DESCRIPTION
**Pull request by muscaln**
Added normal package definiton so `nix-build` also works.

### CODE REVIEW

File: packages/nix/polymc/default.nix [hunk 1]
```diff
@@ -69,11 +96,13 @@ mkDerivation rec {
   };
 
   postInstall = ''
```
Comment (cidkidnix) [inline packages/nix/polymc/default.nix:77]: We need to add a `rm $out/share/applications/*` first here, due to the fact that the Cmake install puts a broken desktop file at `$out/share/applications/org.polymc.PolyMC.desktop` this is why I had a substitute in installPhase for the desktop file to patch the Exec= line

Comment (muscaln) [reply packages/nix/polymc/default.nix:77]: Thanks but check line 100. Already overwriting that.

Comment (cidkidnix) [reply packages/nix/polymc/default.nix:77]: oops didn't see that in the PR, my bad

**Review by bbjubjub2494 — APPROVED** [review]

**Comment by muscaln** [comment]
PR is ready to merge. @cidkidnix @binex-dsk

**Review by cidkidnix — APPROVED** [review]

**Pull request merged**